ARCs/arc-0080.md

```diff
@@ -0,0 +1,145 @@
+And the teal program associated.
+Each storage type has specific requirements.
+
+### Asset Query URI (`algorand:asset://`)
+
+The asset URI enables retrieval of metadata and configuration details for a specific asset, such as its name, total supply, decimal precision, and associated addresses.
+
+### ABNF Grammar
+
+```abnf
+algorandappurn     = "algorand:app://" appid [ "?" noopparams ]
+appid              = *digit
+noopparams         = noopparam [ "&" noopparams ]
+noopparam          = [ boxparam / globalparam / localparam ]
+boxparam           = "box=" *qbase64url
+globalparam        = "global=" *qbase64url
+localparam         = "local=" *qbase64url "&"account=" *base32
+tealcodeparam      = "tealcode"
+
+algorandasseturn   = "algorand:asset://" assetid [ "?" assetparam ]
+assetid            = *digit
+assetparam         = [ totalparam / decimalsparam / frozenparam / unitnameparam / assetnameparam / urlparam / hashparam / managerparam / reserveparam / freezeparam / clawbackparam ]
+totalparam         = "total"
+decimalsparam      = "decimals"
+frozenparam        = "frozen"
+unitnameparam      = "unitname"
+assetnameparam     = "assetname"
+urlparam           = "url"
+hashparam          = "metadatahash"
+managerparam       = "manager"
+reserveparam       = "reserve"
+freezeparam        = "freeze"
+clawbackparam      = "clawback"
+```
+
+- "qchar" corresponds to valid characters of an RFC 3986 URI query component, excluding the "=" and "&" characters, which this specification takes as separators.
+- "qbase64url" corresponds to valid characters of "base64url" encoding, as defined in <a href="https://www.rfc-editor.org/rfc/rfc4648.html#section-5">RFC 4648 section 5</a>
+
+### Parameter Definitions
+
+#### Application Parameters
+
+- **`boxparam`**: Queries the application’s box storage with a key encoded in `base64url`.
+- **`globalparam`**: Queries the global storage of the application using a `base64url`-encoded key.
+- **`localparam`**: Queries local storage for a specified account. Requires an additional `algorandaddress` parameter, representing the account whose local storage is queried.
+- **`account`**: Account to be used in Application NoOp method call foreign accounts array.
+
+#### Asset Parameters
+
+- **`totalparam`** (`total`): Queries the total supply of the asset.
+- **`decimalsparam`** (`decimals`): Queries the number of decimal places used for the asset.
+- **`frozenparam`** (`frozen`): Queries whether the asset is frozen by default.
+- **`unitnameparam`** (`unitname`): Queries the short name or unit symbol of the asset (e.g., "USDT").
+- **`assetnameparam`** (`assetname`): Queries the full name of the asset (e.g., "Tether").
+- **`urlparam`** (`url`): Queries the URL associated with the asset, providing more information.
+- **`hashparam`** (`metadatahash`): Queries the metadata hash associated with the asset.
```

Comment:
I wanted to be cheap on the bytes-size in case it is used at scale.
But now that you are saying it, maybe keeping the same naming convention is more important.